src/Bridge/Client.php

```diff
@@ -2,27 +2,71 @@
 
 namespace Laravel\Passport\Bridge;
 
+use Laravel\Passport\Bridge\ScopedClientInterface;
 use League\OAuth2\Server\Entities\Traits\ClientTrait;
 use League\OAuth2\Server\Entities\Traits\EntityTrait;
-use League\OAuth2\Server\Entities\ClientEntityInterface;
 
-class Client implements ClientEntityInterface
+class Client implements ScopedClientInterface
 {
     use ClientTrait, EntityTrait;
 
+    /**
+     * @var string[]
+     */
+    protected $allowedScopes;
+
     /**
      * Create a new client instance.
      *
-     * @param  string  $identifier
-     * @param  string  $name
-     * @param  string  $redirectUri
+     * @param  string $identifier
+     * @param  string $name
+     * @param  string $redirectUri
+     * @param  mixed  $allowedScopes
      * @return void
      */
-    public function __construct($identifier, $name, $redirectUri)
+    public function __construct($identifier, $name, $redirectUri, $allowedScopes = null)
     {
         $this->setIdentifier($identifier);
+        $this->setAllowedScopes($allowedScopes);
 
         $this->name = $name;
         $this->redirectUri = explode(',', $redirectUri);
     }
+
+    /**
+     * Get the client's allowed scopes.
+     * If this is empty, the client will be able to request
+     * all scopes (no restriction).
+     *
+     * @return array
+     */
+    public function getAllowedScopes()
+    {
+        return array_filter($this->allowedScopes);
+    }
+
+    /**
+     * Set allowed scopes attributes converting optional
+     * param $allowedScopes to array.
+     * If the client has no allowed scopes defined or the string '*',
+     * it means the client has no scope restrictions.
+     *
+     * @param mixed $allowedScopes
+     */
+    protected function setAllowedScopes($allowedScopes = null)
+    {
+        if (!$allowedScopes || '*' === $allowedScopes) {
```

Comment:
This logic to define all scopes (no restrictions) can be encapsulated in a method, because it have a reason to exists that maybe it's not so clear using a simple if, and is duplicated in the explanation (setAllowedScopes() and getAllowedScopes())

The method can have a phpdoc explaining that no rescrition = * or empty... and then those other methods that explain "no scopes or *" can be a simple "no restriction"